### ISSUE TITLE
Tree item focus should avoid jumping when tree items change

### ISSUE THREAD

**Issue by alexr00** [microsoft/vscode#126964]
I have an element A which is focused in the tree. The tree is refreshed. Element A is no longer there. Focus is moved to the tree since the element is missing.
The proper solution would be to keep the focus stable, thus moving it to the neighbor element if the previous one is no longer there

Explorer runs into the same issue, element A is focused and delete it via Delete key. We move focus to the neighbor element, but we can do that since we have our internal api.

**Comment by isidorn** [comment]
fyi @joaomoreno since more and more trees are hitting this, maybe would be cool if focus preservation is done on the tree level, so explorer and custom tree do not have to handle this both separately.

**Comment by isidorn** [comment]
For reference here's a code pointer where Explorer is doing focus manipulation when it knows an item is about to disappear
https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/files/browser/explorerService.ts#L349

**Comment by joaomoreno** [comment]
This should do it. @isidorn I even removed that code from the Explorer, please verify everything works OK. One bug still: deleting the last file from the explorer will throw an exception. Fixing...

**Comment by isidorn** [comment]
@joaomoreno seems to work great for the explorer, thanks a lot for doing this 👏  Last item also seems to work for me
Let me add bug label so I verify this one more time in endgame.
Live share will also try it out for their use case on Monday.

**Comment by joaomoreno** [comment]
Deleting the last file fix: https://github.com/microsoft/vscode/commit/5527c69c4bc6244fe1962496d36f40278f3c3666

**Comment by joaomoreno** [comment]
@alexr00 https://github.com/microsoft/vscode/pull/201309 will revert behavior. Let's chat soon in how we can address this specifically for custom tree views.

**Comment by alexr00** [comment]
FYI @lramos15 too since this will affect the explorer.

**Comment by joaomoreno** [comment]
> FYI @lramos15 too since this will affect the explorer.

https://github.com/microsoft/vscode/pull/201309 makes sure the Explorer doesn't regress.
